### PR TITLE
Ben/long covid updates

### PR DESCRIPTION
**Pull request by bionara**
# Description

Addresses multi profile for long covid survey
Address PR comments around best practices

## On what platform have you tested the change?

- [ ] Android - Emulator
- [ ] Android - Physical device
- [x] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots

_Please attach screenshots showing the change if relevant_

## Checklist

- [ ] I have updated mockServer
- [ ] I've added analytics as relevant
- [ ] I have run `npm test`
- [ ] I have run `npm run test:i18n`

## Out of scope and potential follow-up

_Is there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?_

### CODE REVIEW

File: src/features/assessment/HowYouFeelScreen.tsx [hunk 1]
```diff
@@ -51,16 +51,16 @@ export const HowYouFeelScreen: React.FC<Props> = ({ route, navigation }) => {
     if (isSubmitting) {
       return;
     }
-    if (!longCovid.had_covid) {
+    if (!longCovidApiClient.dataContainsPatientId(longCovidState.profiles, assessmentCoordinator.assessmentData.patientData.patientId)) {
```
Comment (selvach) [inline src/features/assessment/HowYouFeelScreen.tsx:54]: You could return this on the patient endpoint... 

When the reporting flow starts we call  GET /patient/xxx-yyy-zz/
On this we could simply return a new parameter `had_completed_long_covid` ?

WDYT?

Comment (bionara) [reply src/features/assessment/HowYouFeelScreen.tsx:54]: Yep that's a grand idea - will update

File: src/features/long-covid/LongCovidApiClient.ts [hunk 2]
```diff
@@ -28,8 +28,10 @@ export class LongCovidApiClient implements ILongCovidApiClient {
...
   }
 
-  update(LongCovid: ILongCovid): Promise<ILongCovid> {
-    const url = `${API_URL}${LongCovid.id}/`;
-    return this.apiClient.patch<ILongCovid, ILongCovid>(url, LongCovid);
+  dataContainsPatientId(longCovidList:ILongCovid[], patientId: string): boolean {
+    if (!longCovidList || !longCovidList.length) {
+      return false;
+    }
+    return longCovidList.filter((longCovid: ILongCovid) => longCovid.patient === patientId).length > 0;
```
[the remark this answers is not in the document]
Comment (bionara) [reply src/features/long-covid/LongCovidApiClient.ts:35]: good shout - updated!

**Review by aikewoody — DISMISSED** [review]
Refactor Ben is taking form, love it!

File: src/features/assessment/HowYouFeelScreen.tsx (hunk 1 above)
Comment (bionara) [reply src/features/assessment/HowYouFeelScreen.tsx]: Done, and removed state. much nicer!

File: src/features/long-covid/LongCovidApiClient.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (bionara) [reply src/features/long-covid/LongCovidApiClient.ts]: Actually removed this in the end in favour of the BE flag!

**Review by selvach — APPROVED** [review]

**Pull request merged**